### ISSUE TITLE
npm@3 fails to install serialport@2, but npm@2 works

### ISSUE THREAD

**Issue by aredridel** [npm/npm#10026]
```
serialport-2@1.0.0 /Users/aredridel/Projects/serialport-2
├── async@0.9.0  extraneous
├── bindings@1.2.1  extraneous
├── debug@2.2.0  extraneous
├── nan@2.0.9  extraneous
├── optimist@0.6.1  extraneous
└── sf@0.1.7  extraneous

npm WARN EPACKAGEJSON serialport-2@1.0.0 No repository field.
npm ERR! Darwin 15.0.0
npm ERR! argv "/usr/local/lib/node_modules/node-darwin-x64/bin/node" "/usr/local/bin/npm" "i" "--save" "serialport@2"
npm ERR! node v4.1.0
npm ERR! npm  v3.3.9
npm ERR! path /Users/aredridel/Projects/serialport-2/node_modules/serialport/node_modules/node-pre-gyp/node_modules/request/node_modules/har-validator/bin/har-validator
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall chmod

npm ERR! enoent ENOENT: no such file or directory, chmod '/Users/aredridel/Projects/serialport-2/node_modules/serialport/node_modules/node-pre-gyp/node_modules/request/node_modules/har-validator/bin/har-validator'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/aredridel/Projects/serialport-2/npm-debug.log
```

debug log at https://gist.github.com/aredridel/c1284461af09f4cf47c7

**Comment by iarna** [comment]
I can't reproduce this in `3.3.9` or `3.3.10`

**Comment by nemrod** [comment]
@iarna this happened to me last week but I've already downgraded so I'm not sure exactly what version I had nor can I post debug logs. Sorry.

**Comment by iarna** [comment]
@nemrod specifically with `serialport@2` or with some other modules?

**Comment by nemrod** [comment]
@iarna: ``serialport@2`` indeed

**Comment by XeCycle** [comment]
`3.3.9` is still failing to install it in my Gitlab CI.  It is a private project depending on `serialport@2` so sorry I cannot post debug logs.

**Comment by XeCycle** [comment]
A test of only `npm -g i serialport@2` succeeded, but installing all dependencies failed.  Can you try a `package.json` with these dependencies?

```
  "dependencies": {
    "URIjs": "1.16.1",
    "babel-core": "5.8.25",
    "baconjs": "0.7.77",
    "bluebird": "2.10.2",
    "event-stream": "3.3.2",
    "html-to-text": "1.3.2",
    "immutable": "3.7.5",
    "joi": "6.9.1",
    "js-quantities": "1.5.0",
    "js-yaml": "3.4.3",
    "koa": "1.1.0",
    "koa-body-parser": "1.1.2",
    "koa-compress": "1.0.8",
    "koa-conditional-get": "1.0.3",
    "koa-dispatcher": "*",
    "koa-etag": "2.0.0",
    "koa-static": "1.5.1",
    "moment": "2.10.6",
    "nodemailer": "1.8.0",
    "path-to-regexp": "1.2.1",
    "pg.js": "4.1.1",
    "react": "0.13.3",
    "request": "2.65.0",
    "serialport": "2.0.1",
    "source-map-support": "0.3.2",
    "ws": "0.8.0"
  },
  "devDependencies": {
    "babel-eslint": "4.1.3",
    "babel-loader": "5.3.2",
    "babel-runtime": "5.8.25",
    "chai": "3.3.0",
    "chai-as-promised": "5.1.0",
    "classnames": "2.2.0",
    "css-loader": "0.20.1",
    "eslint": "1.7.1",
    "eslint-loader": "1.1.0",
    "exports-loader": "0.6.2",
    "file-loader": "0.8.4",
    "imports-loader": "0.6.5",
    "jquery": "2.1.3",
    "json-loader": "0.5.3",
    "mocha": "2.3.3",
    "parse-messy-time": "1.2.1",
    "pegjs-loader": "0.1.1",
    "raw-loader": "0.5.1",
    "sinon": "1.17.1",
    "sinon-chai": "2.8.0",
    "style-loader": "0.12.4",
    "url-loader": "0.5.6",
    "webpack": "1.12.2"
  }
```

**Comment by XeCycle** [comment]
`npm-debug.log` with a `package.json` containing only the above dependencies: https://gist.github.com/XeCycle/d727d24f86e1990bcb22

**Comment by iarna** [comment]
@XeCycle THANK YOU! This repros for me! We may finally track this down! (It's show up intermittently with other installations, but never in a way anyone could consistently reproduce!)

**Comment by simonua** [comment]
@iarna Awesome fix! The ENOENT issue gave us quite a bit of headache prior to 3.3.11 but appears to be working well now. Thank you very much to everyone involved from troubleshooting to fix!

**Comment by iarna** [comment]
The fix for this landed in 3.3.11! =)

**Comment by Dignifiedquire** [comment]
I'm afraid this issue is not fully resolved, still seeing it on `3.3.12`. 

```
npm ERR! Darwin 15.0.0
npm ERR! argv "/Users/dignifiedquire/.nvm/versions/node/v4.1.2/bin/node" "/Users/dignifiedquire/.nvm/versions/node/v4.1.2/bin/npm" "update"
npm ERR! node v4.1.2
npm ERR! npm  v3.3.12
npm ERR! path /Users/dignifiedquire/opensource/ipfs/webui/node_modules/bygg-plugins-light/node_modules/stream-combiner2/node_modules/readable-stream
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall rename

npm ERR! enoent ENOENT: no such file or directory, rename '/Users/dignifiedquire/opensource/ipfs/webui/node_modules/bygg-plugins-light/node_modules/stream-combiner2/node_modules/readable-stream' -> '/Users/dignifiedquire/opensource/ipfs/webui/node_modules/bygg-plugins-light/node_modules/module-deps/node_modules/stream-combiner2/node_modules/readable-stream'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent
```

It happens to me only when changing dependencies (adding or updating) and then running `npm install`. Most of the times running `npm prune && npm install` fixes it, but sometimes I have to `rm -rf node_modules` to get back into a working state. I overwrote the last debug log by accident but will post as soon as it happens again.

**Comment by Dignifiedquire** [comment]
Was able to reproduce it again, this is the error message:

```
npm WARN EPACKAGEJSON webterm@1.0.0 No repository field.
npm ERR! Darwin 15.0.0
npm ERR! argv "/Users/dignifiedquire/.nvm/versions/node/v4.1.2/bin/node" "/Users/dignifiedquire/.nvm/versions/node/v4.1.2/bin/npm" "update"
npm ERR! node v4.1.2
npm ERR! npm  v3.3.8
npm ERR! path /Users/dignifiedquire/opensource/node-webterm/node_modules/socket.io/node_modules/engine.io/node_modules/engine.io-parser/node_modules/after
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall rename

npm ERR! enoent ENOENT: no such file or directory, rename '/Users/dignifiedquire/opensource/node-webterm/node_modules/socket.io/node_modules/engine.io/node_modules/engine.io-parser/node_modules/after' -> '/Users/dignifiedquire/opensource/node-webterm/node_modules/after'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/dignifiedquire/opensource/node-webterm/npm-debug.log
```

and here is the [npm-debug.log](https://gist.github.com/Dignifiedquire/0124824df21a488b098f).
